### PR TITLE
SC-419: How to for RBAC

### PR DESCRIPTION
**Pull request by harrycguo**
### Your checklist for this pull request

Thank you for your contribution to the `next-dev-docs` repo. 

🚨Please review the [Pull request guidelines](../contributor-guide/contributor-guide.md#pull-request-guidelines) to this repository before submitting this PR.

- [ ] Your code builds clean without any errors or warnings
  
  ```bash
  yarn build
  ```

- [ ] You are using approved terminology

  Refer to the [style guide](../contributor-guide/style-guide.md) for help.

### Description 

JIRA Story: https://beyondidentity.atlassian.net/browse/SC-419

I moved some things around, notably:

To top level "How to Guides":

- Create a role
- Add users and groups to a role
- Configure RBAC

To the "Administrative" section:
- Configure Administrative Roles

### CODE REVIEW

File: docs/how-to/configure-rbac.mdx [hunk 1]
```diff
@@ -1,29 +1,79 @@
...
+to access a certain part of your site, you can use RBAC to allow or deny access.
+
+Now, let's identify the different components that make up RBAC and walk through the example above.
+
+## Resource Servers and Scopes
+
+Resource servers represent an entity hosting protected resources that the client is attempting to access. This is the entire set of permissions that are allowed on a resource.
+These permissions are expressed through scopes, which are strings that represent the set of permissions.
+
+For our example above, we would create a resource server called Shoes and the scopes that are available could be `shoes:read`, `shoes:buy`, `clearance:buy`, `new-arrivals:read`, `new-arrivals:buy`, etc.
```
Comment (Will5) [inline docs/how-to/configure-rbac.mdx:37]: We should probably link to the "Add a resource server" doc.

File: docs/how-to/configure-rbac.mdx [hunk 2]
```diff
@@ -1,29 +1,79 @@
...
+1. Scopes on the `/authorize` request (during the OAuth2 / OIDC flow).
+1. Scopes supported by the Application the identity is authenticating into.
+1. Scopes tied to the identity's roles.
+
+For our example above, a free member authenticates into the website would have an access token with the scopes `shoes:read`, `shoes:buy`, `clearance:buy`, and a VIP
+member's access token would contain the scopes `shoes:read`, `shoes:buy`, `clearance:buy`, `new-arrivals:read`, `new-arrivals:buy`.
+
+## Token Introspection
+
+Whenever a [token introspection](<link to token introspection>) is called for an access token with an identity as the principal, the scopes returned in the introspect
```
Comment (turnipdabeets) [inline docs/how-to/configure-rbac.mdx:63]: need link here :)

File: docs/how-to/configure-rbac.mdx [hunk 3]
```diff
@@ -1,29 +1,79 @@
...
+response will contain **only a subset of the scopes for which the identity has the role(s) for**.
```
Comment (turnipdabeets) [inline docs/how-to/configure-rbac.mdx:64]:
```suggestion
When invoking [token introspection](<link to token introspection>) for an access token associated with an identity as the principal, the returned scopes in the introspection response will contain **only a subset of the scopes for which the identity has the role(s) for**.
```

File: docs/how-to/configure-rbac.mdx [hunk 4]
```diff
@@ -1,29 +1,79 @@
...
+the new arrivals, the introspect response should NOT return the `new-arrivals:read` scope on the response and the user should be denied access to that page.
+
+## Putting it all together
+
+After creating the aforementioned resource server and roles above, we would create an application with the authorization code flow and integrate that with your application.
+Now, whenever a user logs in and attempts to access the various parts of your website, RBAC will ensure that the logged in user can only access the resources that are allowed
+to them by their role.
+
+RBAC is an authorization tool that can be used to validate access permissions on the end user, and is just one of the tools that Beyond Identity can help you accomplish your
+identity and access management goals.
```
Comment (turnipdabeets) [inline docs/how-to/configure-rbac.mdx:79]:
```suggestion
After creating the aforementioned resource server and roles above, the next step involves the creation of an application employing the authorization code flow using the roles and scopes created. Consequently, whenever a user logs in and attempts to access different sections of your website, Role-Based Access Control (RBAC) will ensure that the logged in user can only access the resources that are allowed to them by their designated role.

RBAC is an authorization tool that can be used to validate access permissions on the end user, and is just one of the tools that Beyond Identity can help you accomplish your identity and access management goals.
```

File: docs/how-to/configure-rbac.mdx [hunk 5]
```diff
@@ -1,29 +1,79 @@
...
+For our example above, say a VIP member logs into the website and halfway through their session, they lose their VIP status. Now when the (now free) member attempts to view
+the new arrivals, the introspect response should NOT return the `new-arrivals:read` scope on the response and the user should be denied access to that page.
+
+## Putting it all together
+
+After creating the aforementioned resource server and roles above, we would create an application with the authorization code flow and integrate that with your application.
+Now, whenever a user logs in and attempts to access the various parts of your website, RBAC will ensure that the logged in user can only access the resources that are allowed
+to them by their role.
+
+RBAC is an authorization tool that can be used to validate access permissions on the end user, and is just one of the tools that Beyond Identity can help you accomplish your
```
Comment (turnipdabeets) [inline docs/how-to/configure-rbac.mdx:78]: Is there a logical next place to link a user for more tools to accomplish identity and access management goals?

File: docs/how-to/configure-rbac.mdx [hunk 6]
```diff
@@ -1,29 +1,79 @@
...
+
+For the most part, the scopes in the token introspection response will match the scopes on the token itself unless permissions change. But should a user's roles change,
+the introspect response will reflect those changes.
+
+For our example above, say a VIP member logs into the website and halfway through their session, they lose their VIP status. Now when the (now free) member attempts to view
+the new arrivals, the introspect response should NOT return the `new-arrivals:read` scope on the response and the user should be denied access to that page.
+
+## Putting it all together
+
+After creating the aforementioned resource server and roles above, we would create an application with the authorization code flow and integrate that with your application.
```
Comment (turnipdabeets) [inline docs/how-to/configure-rbac.mdx:74]: Is there anything we should point out in how to integrate that with the application?

**Review by turnipdabeets — COMMENTED** [review]
Looks good! A couple of rewordings and a few questions.

File: docs/how-to/configure-rbac.mdx (hunk 5 above)
Comment (harrycguo) [reply docs/how-to/configure-rbac.mdx]: I don't think so, this is it!

File: docs/how-to/configure-rbac.mdx (hunk 6 above)
Comment (harrycguo) [reply docs/how-to/configure-rbac.mdx]: Hmmm I'm not sure if there is anything besides this step. The developer must take steps to introspect within their own application after this.

**Review by Will5 — APPROVED** [review]

**Comment by harrycguo** [comment]
@Will5 @turnipdabeets I added this to the PR description, but I moved some things around, notably:

To top level "How to Guides":

- Create a role
- Add users and groups to a role
- Configure RBAC

To the "Administrative" section I added:
- Configure Administrative Roles (basic page on configuring admin roles)

**Pull request merged**